### PR TITLE
testing: #47 Add unit tests

### PR DESCRIPTION
**Pull request by abhijit-hota**
**Description**
- Add and configure Jest and other related dependencies to start implementing unit tests.
- Add basic unit test for the `Title` component.

**Related issue**: #47 

**References**
- https://www.gatsbyjs.com/docs/unit-testing/ (Mainly followed this instruction step by step)
- https://github.com/gatsbyjs/gatsby/tree/master/examples/using-jest
- https://jestjs.io/docs/en/snapshot-testing.html

**Note:**
- The directory structure is the recommended one in the above links. Thought of moving the three extra files (except `jest.config.js`) as they crowd the root folder a bit but I didn't want to deviate from the recommended structure much.
- [Jest recommends committing snapshot files](https://jestjs.io/docs/en/snapshot-testing.html#should-snapshot-files-be-committed).

Awaiting review. Thanks! 😄

### CODE REVIEW

**Comment by abhijit-hota** [comment]
If the changes get approved, I'll make the necessary changes to the Readme to have the instructions for testing and writing tests.

(Also will make a comment to all-contributors for `test` 🙈 )

File: src/components/__tests__/__snapshots__/title.test.js.snap [hunk 1]
```diff
@@ -0,0 +1,30 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
```
Comment (rahuldkjain) [inline src/components/__tests__/__snapshots__/title.test.js.snap:1]: Is it necessary to keep snapshots in the repository?

Comment (abhijit-hota) [reply src/components/__tests__/__snapshots__/title.test.js.snap:1]: I researched about the same before committing. As I've said above, [Jest ~recommends~ insists on keeping them](https://jestjs.io/docs/en/snapshot-testing.html#should-snapshot-files-be-committed).

My understanding: Snapshots are required by Jest so that it can match the generated tree with them. If not available, snapshots are generated again by Jest for running tests. **This can slow down the tests when you run them for the first time** in your machine.

Bottom line is, having snapshots in the repo can increase the pace of testing, for new users. So my opinion is to keep them if we plan to implement more tests.

Comment (rahuldkjain) [reply src/components/__tests__/__snapshots__/title.test.js.snap:1]: Oh Cool. Then we can keep it.

**Review by rahuldkjain — APPROVED** [review]
Superb @abhijit-hota.

**Comment by abhijit-hota** [comment]
> Superb @abhijit-hota.

Thanks! I can learn more about Jest and add more tests if time permits me 👍.
